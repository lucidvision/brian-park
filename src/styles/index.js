export const colors = {
  primary: 'rgba(0, 13, 85, 0.8)',
  black: '#000',
  white: '#fff',
  gradientLight: 'rgba(0, 38, 255, 0.8)',
  gradientDark: 'rgba(0, 0, 0, 0.8)'
}

export const fontSizes = {
  large: '6rem',
  medium: '4rem',
  small: '2.2rem',
  default: '1.6rem'
}

export const screenSizes = {
  desktop: '112.5em',
  tablet: '75em',
  phone: '56.25em'
}

export const utilities = {
  marginSmall: '1.5rem',
  marginMedium: '4rem',
  marginLarge: '8rem'
}
