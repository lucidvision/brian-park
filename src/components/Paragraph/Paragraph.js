import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'
import { colors, fontSizes } from 'styles'

Paragraph.propTypes = {
  color: PropTypes.string,
  size: PropTypes.string,
  text: PropTypes.string.isRequired
}

Paragraph.defaultProps = {
  color: colors.primary,
  size: fontSizes.default,
  text: ''
}

const StyledParagraph = styled.p`
  color: ${props => props.color};
  font-size: ${props => props.size};
`

export default function Paragraph (props) {
  const { text, ...rest } = props
  return (
    <StyledParagraph {...rest}>{text}</StyledParagraph>
  )
}
