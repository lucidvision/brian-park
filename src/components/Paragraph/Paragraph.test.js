import React from 'react'
import { mount } from 'enzyme'
import Paragraph from './Paragraph'
import { colors, fontSizes } from 'styles'

const createParagraph = (override) => {
  const props = {
    text: 'Unicorns',
    ...override
  }

  return mount(<Paragraph {...props} />)
}

describe('Paragraph', () => {
  const component = createParagraph()

  it('renders correctly', () => {
    expect(component).toMatchSnapshot()
  })

  it('renders with the correct props', () => {
    expect(component.find('Paragraph').props()).toEqual({
      color: colors.primary,
      size: fontSizes.default,
      text: 'Unicorns'
    })
  })

  it('renders p with the correct text', () => {
    expect(component.find('p').props().children).toBe('Unicorns')
  })
})
