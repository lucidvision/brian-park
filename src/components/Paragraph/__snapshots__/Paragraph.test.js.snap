// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Paragraph renders correctly 1`] = `
<Paragraph
  color="rgba(0, 13, 85, 0.8)"
  size="1.8rem"
  text="Unicorns"
>
  <styled.p
    color="rgba(0, 13, 85, 0.8)"
    size="1.8rem"
  >
    <p
      className="sc-bdVaJa HuCCv"
      color="rgba(0, 13, 85, 0.8)"
      size="1.8rem"
    >
      Unicorns
    </p>
  </styled.p>
</Paragraph>
`;
